pylint/checkers/classes.py

```diff
@@ -758,6 +834,27 @@ def _check_useless_super_delegation(self, function):
         if super_call.type.name != current_scope.name:
             return
 
+        # Check values of default args
+        klass = function.parent.frame()
+        for overridden in klass.local_attr_ancestors(function.name):
+            # get astroid for the searched method
+            try:
+                meth_node = overridden[function.name]
+            except KeyError:
+                # we have found the method but it's not in the local
+                # dictionary.
+                # This may happen with astroid build from living objects
+                continue
+            if not isinstance(meth_node, astroid.FunctionDef):
+                # If the method have an ancestor which is not a function
+                # then it is legitimate to redefine it
+                return
+            if not _has_different_parameters_default_value(meth_node.args, function.args):
+                self.add_message('useless-super-delegation', node=function, args=(function.name,))
+            # For each method that have an ancestor, the previous check is enough
```

Comment:
I don't think this is correct. We still need to check that the call and the definition are equivalent, since this change only helps with default values. Here's an example where your code will emit a ``useless-super-delegation`` even though the super() call is not equivalent.

```
class A:
   def test(self, a=None):
      pass


class B(A):
   def test(self, a=None):
       super(B, self).test(a=a+1)
```

From what I see, if it these two functions have different default values, we should just skip the ancestor. If all ancestors behave the same, bail out and don't check for useless super delegation any longer. Otherwise, go to the signature check

Reply:
@PCManticore you are absolutely right. I will correct this ASAP. Thanks.